site/content/docs/02-template-syntax.md

```diff
@@ -887,7 +887,140 @@ Local transitions only play when the block they belong to is created or destroye
 
 ### Animations
 
-TODO i can't remember how any of this works
+```sv
+animate:name
+```
+
+```sv
+animate:name={params}
+```
+
+```js
+animation = (node: HTMLElement, { from: DOMRect, to: DOMRect } , params: any) => {
+	delay?: number,
+	duration?: number,
+	easing?: (t: number) => number,
+	css?: (t: number, u: number) => string,
+	tick?: (t: number, u: number) => void
+}
+```
+
+```js
+DOMRect {
+	bottom: number,
+	height: number,
+	​​left: number,
+	right: number,
+	​top: number,
+	width: number,
+	x: number,
+	y:number
+}
+```
+
+---
+
+An animation is triggered when the contents of a [keyed each block](docs#Each_blocks) are re-ordered. Animations do not run when an element is removed, only when the each block's data is reordered. Animate directives must be on an element that is an *immediate* child of a keyed each block.
+
+Animations will normally be used with the [`flip`](docs#flip) function which will swap the positions of elements when their order changes.
+
+```html
+<!-- When `list` is reordered the animation will run-->
+{#each list as item, index (item)}
+	<li animate:flip>{item}</li>
+{/each}
+```
+
+#### Animation Parameters
+
+---
+
+As with actions and transitions, animations can have parameters.
+
+(The double `{{curlies}}` aren't a special syntax; this is an object literal inside an expression tag.)
+
+```html
+{#each list as item, index (item)}
+	<li animate:flip="{{ delay: 500 }}">{item}</li>
+{/each}
+```
+
+#### Custom animation functions
+
+---
+
+Animations can use custom functions, these are very similar to custom transition functions. Custom animations also give you access to an `animation` parameter which is an object containing `from` and `to` properties containing the [DOMRect](https://developer.mozilla.org/en-US/docs/Web/API/DOMRect#Properties) (geometry) of the element. The `from` property is the geometry of the element in its starting position, the `to` property is the geometry of the element after the list has been re-ordered and the DOM updated.
+
+If the returned object has a css function, Svelte will create a CSS animation that plays on the element.
+
+The `t` argument passed to css is a value between 0 and 1 after the easing function has been applied. `t` always starts with a value of 0 and ends with a value of 1. `u` always starts with a value of 1 and ends with a value of 0.
```

Comment:
If this was copy-pasted from the transitions section, it lost some formatting. I'm not sure it's necessary to specifically mention that ranges of both t and u.

Maybe:

```suggestion
The `t` argument passed to `css` is a value that goes from `0` and `1` after the `easing` function has been applied. The `u` argument is equal to `1 - t`.
```